SimplenoteWidgets/Tools/WidgetDataController.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import SimplenoteFoundation
+import SimplenoteSearch
+
+class WidgetDataController {
+
+    /// Data Controller
+    ///
+    let coreDataManager: CoreDataManager
+
+    /// Notes Controller
+    ///
+    private lazy var notesController = ResultsController<Note>(
+        viewContext: coreDataManager.managedObjectContext,
+        matching: predicateForNotes(),
+        sortedBy: [NSSortDescriptor.descriptorForNotes(sortMode: WidgetDefaults.shared.sortMode)]
+    )
+
+    /// Initialization
+    ///
+    init(coreDataManager: CoreDataManager, isPreview: Bool = false) throws {
+        if !isPreview {
+            guard WidgetDefaults.shared.loggedIn else {
+                throw WidgetError.appConfigurationError
+            }
+        }
+
+        self.coreDataManager = coreDataManager
+    }
+
+    private func performFetch() throws {
+        do {
+            try notesController.performFetch()
+        } catch {
+            throw WidgetError.fetchError
+        }
+    }
+
+
+    // MARK: Public Methods
+
+    /// Fetch notes with given tag and limit
+    /// If no tag is specified, will fetch notes that are not deleted. If there is no limit specified it will fetch all of the notes
+    ///
+    func notes(withTag tag: String? = nil, limit: Int = .zero) throws -> [Note] {
+        notesController.limit = limit
+
+        if let tag = tag {
+            notesController.predicate = predicateForNotes(withTag: tag)
+        }
+        try performFetch()
+
+        return notesController.fetchedObjects
+    }
+
+    /// Returns note given a simperium key
+    ///
+    func note(forSimperiumKey key: String) -> Note? {
+        notesController.predicate = predicateForNotes()
```

Comment:
Let's chat over Slack, I believe we don't really need the whole ResultsController / NSFetchedResultsController machinery

Reply:
Quick sketch, but the same machinery, without FRC, would look as follows:

```
        let request = NSFetchRequest<Note>(entityName: Note.entityName)
        request.predicate = predicate
        request.sortDescriptors = sortDescriptors
        request.fetchLimit = limit

        // This should be an ivar?
        let mainContext = coreDataManager.managedObjectContext!
        mainContext.fetch(request)
```